.gitignore

```diff
@@ -8,7 +8,8 @@
 node_modules/
 npm-debug.log
 yarn-error.log
-  
+package-lock.json
+yarn.lock
```

Comment:
:exclamation: lock files should not be ignored
```suggestion
```

Reply:
Why? Do you want to have merge conflicts on generic output?

Reply:
git/npm (not sure which one) is smart enough to resolve the merge conflicts so I really doubt merge conflict will ever happen (it has never happened to me so far)

It is important to have lock file in git so that installation so that anyone who clones the project gets the exact same version

Reply:
It will be a vice versa, npm + git will create these conflicts.
I can simulate simple situation:
1) 2 developers do clone. (for example ts lib: `4.0.4`)
2) after 2 weeks first developer will create pr. He did just `npm i` - and package-lock.json is updated now. Because some of libraries have updates (`4.1.0`)
3) after 3 weeks, another one repeat it (without merged pr of step 2). `package-lock.json` will be different, because there was some new update of library. (`4.1.2`)

Congrats, they have merge conflict.

This file is useful for project with big dependency graph. So I strongly don't recommend to use it for library with almost zero dependencies.

So decide.

Reply:
I understand your concern but I really doubt we will end up in this scenario.
And even if it happens there should be no problem to resolve the conflicts - I can easily handle that.